processing/src/main/java/org/apache/druid/query/operator/SegmentToRowsAndColumnsOperator.java

```diff
@@ -48,6 +48,9 @@ public Closeable goOrContinue(Closeable continuation, Receiver receiver)
       }
 
       RowsAndColumns rac = shifty.as(RowsAndColumns.class);
+      if (shifty instanceof RowsAndColumns) {
+        rac = (RowsAndColumns) shifty;
```

Comment:
Shouldn't `shifty` have returned `this` from `as` in this case? Wonder why we need this block.

Reply:
This if statement is definitely placed quite weird-like, you are right